src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/DataCenterFormula.kt

```diff
@@ -127,11 +129,34 @@ class DataCenterFormula private constructor(
     ): ProvisionedJira = time("provision Jira Data Center") {
         logger.info("Setting up Jira...")
 
-        val executor = Executors.newCachedThreadPool(
-            ThreadFactoryBuilder()
-                .setNameFormat("data-center-provisioning-thread-%d")
-                .build()
-        )
+        AbruptExecutorService(
+            Executors.newCachedThreadPool(
+                ThreadFactoryBuilder()
+                    .setNameFormat("data-center-provisioning-thread-%d")
+                    .build()
+            )
+        ).use { executor ->
+            provision(
+                executor = executor,
+                investment = investment,
+                pluginsTransport = pluginsTransport,
+                resultsTransport = resultsTransport,
+                key = key,
+                roleProfile = roleProfile,
+                aws = aws
+            )
+        }
+    }
+
+    private fun provision(
```

Comment:
Thanks for minimizing the diff :heart:

Reply:
Hopefully we will reduce the length of this method soon, so that such diff minimalisation won't be necessary.